### ISSUE TITLE
build error / gcc /fedora rawhide

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#813]
```
  CC      src/dns.o
In file included from src/http_htx.c:27:
In function 'regex_exec2',
    inlined from '__http_find_header' at src/http_htx.c:171:9:
include/haproxy/regex.h:106:18: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
  106 |  subject[length] = 0;
      |  ~~~~~~~~~~~~~~~~^~~
In file included from include/haproxy/http_htx-t.h:31,
                 from include/haproxy/http_htx.h:28,
                 from src/http_htx.c:24:
src/http_htx.c: In function '__http_find_header':
include/haproxy/htx-t.h:221:7: note: at offset 0 to object 'blocks' with size 0 declared here
  221 |  char blocks[0] __attribute__((aligned(8)));
      |       ^~~~~~
  CC      src/raw_sock.o
cc1: all warnings being treated as errors
```

**Comment by chipitsine** [comment]
@TimWolla , after https://github.com/haproxy/haproxy/commit/4987a474469d6c7c50ea2a8997cdd79f5059cb0c
fedora 33 is happy

should we close this issue ?

**Comment by TimWolla** [comment]
The commit does not contain backporting info. Let's rather not close this for now in case it needs to be backported.

@wtarreau

**Comment by wtarreau** [comment]
I first noticed it only on cygwin just after applying your calloc fix while it's totally unrelated, thus I suspected that maybe the default compiler had just been upgraded on cygwin. I just didn't notice that it had also been reported on fedora earlier, likely as new options appeared with the upgrade to gcc 10. I didn't intend to backport it, but if 2.2 also fails on fedora 33, then we can backport it.
